bin/inx-chronicle/src/api/routes.rs

```diff
@@ -0,0 +1,119 @@
+    }
+
+    Router::new()
+        .nest("/api", router)
+        .fallback(not_found.into_service())
+        .layer(Extension(db))
+        .layer(CatchPanicLayer::new())
+        .layer(TraceLayer::new_for_http())
+        .layer(
+            CorsLayer::new()
+                .allow_origin(Any)
+                .allow_methods(vec![Method::GET, Method::OPTIONS])
+                .allow_headers(Any)
+                .allow_credentials(true),
+        )
+}
+
+async fn info() -> InfoResponse {
+    let version = std::env!("CARGO_PKG_VERSION").to_string();
+    let is_healthy = true;
+    InfoResponse {
+        name: "Chronicle".into(),
+        version,
+        is_healthy,
+    }
+}
+
+async fn sync(database: Extension<Database>) -> APIResult<SyncDataResponse> {
+    let mut res = database
+        .collection::<SyncRecord>("sync")
+        .find(
+            doc! { "synced": true },
+            FindOptions::builder().sort(doc! {"milestone_index": 1}).build(),
+        )
+        .await?;
+    let mut sync_data = SyncData::default();
+    let mut last_record: Option<SyncRecord> = None;
+    while let Some(sync_record) = res.try_next().await? {
+        // Missing records go into gaps
+        if let Some(last) = last_record.as_ref() {
+            if last.milestone_index + 1 != sync_record.milestone_index {
+                sync_data
+                    .gaps
+                    .push(last.milestone_index + 1..sync_record.milestone_index - 1);
+            }
+        }
+        // Synced AND logged records go into completed
+        if sync_record.logged {
+            match sync_data.completed.last_mut() {
+                Some(last) => {
+                    if last.end + 1 == sync_record.milestone_index {
+                        last.end += 1;
+                    } else {
+                        sync_data
+                            .completed
+                            .push(sync_record.milestone_index..sync_record.milestone_index);
+                    }
+                }
+                None => sync_data
+                    .completed
+                    .push(sync_record.milestone_index..sync_record.milestone_index),
+            }
+        // Otherwise the are synced only
+        } else {
+            match sync_data.synced_but_unlogged.last_mut() {
+                Some(last) => {
+                    if last.end + 1 == sync_record.milestone_index {
+                        last.end += 1;
+                    } else {
+                        sync_data
+                            .synced_but_unlogged
+                            .push(sync_record.milestone_index..sync_record.milestone_index);
+                    }
+                }
+                None => sync_data
+                    .synced_but_unlogged
+                    .push(sync_record.milestone_index..sync_record.milestone_index),
+            }
+        }
+        last_record.replace(sync_record);
+    }
+    Ok(SyncDataResponse(sync_data))
```

Comment:
At some point we should move this logic into a separate place to make it easier to write test cases for it. I have [created an issue](https://github.com/iotaledger/inx-chronicle/issues/58).